client/config/config.go

```diff
@@ -485,6 +507,53 @@ func (c *ClientTemplateConfig) IsEmpty() bool {
 		c.NomadRetry.IsEmpty()
 }
 
+func (c *ClientTemplateConfig) Merge(o *ClientTemplateConfig) *ClientTemplateConfig {
```

Comment:
We have https://github.com/hashicorp/nomad/blob/main/helper/pointer/pointer.go#L31-L37 to help clean up these pointer merges